org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -888,6 +941,25 @@ public Preferences setJavaCompletionFavoriteMembers(List<String> javaCompletionF
 		return this;
 	}
 
+	public Preferences putSha256(List<String> sha256Allowed, List<String> sha256Disallowed) {
+		List<String> oldAllowed = this.sha256Allowed;
+		List<String> oldDisallowed = this.sha256Disallowed;
+		WrapperValidator.getAllowed().clear();
+		this.sha256Allowed = sha256Allowed;
+		WrapperValidator.getDisallowed().clear();
```

Comment:
no, getDisallowed() should be immutable

Reply:
there should be a WrapperValidator.clear() method that encapsulates allowed.clear() &  disallowed.clear()

Reply:
Fixed.